.github/workflows/provision.yml

```diff
@@ -42,8 +43,113 @@ on:
+    outputs:
+      backup-ssh-key: ${{ steps.fetch-credentials.outputs.backup-ssh-key }}
+    steps:
+      - name: Fetch backup SSH key
+        id: fetch-credentials
+        run: |
+          SSH_KEY="${{ secrets.SSH_KEY }}"
+          echo -n "$SSH_KEY" | openssl enc -aes-256-cbc -pbkdf2 -salt -k "${{ secrets.GH_ENCRYPTION_PASSWORD }}" -out encrypted_key.bin
+          ENCODED_ENCRYPTED_SECRET=$(base64 < encrypted_key.bin)
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "backup-ssh-key<<$EOF" >> $GITHUB_OUTPUT
+          echo "$ENCODED_ENCRYPTED_SECRET" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
+
+  check-production-environment:
+    name: Check if production environment exists
+    if: ${{ github.event.inputs.environment == 'staging' }}
+    runs-on: ubuntu-22.04
+    outputs:
+      production_env_exists: ${{ steps.check-env.outputs.exists }}
+    steps:
+      - name: Check if GITHUB_TOKEN is set
+        id: check-token
+        run: |
+          if [ -z "${{ secrets.GITHUB_TOKEN }}" ]; then
+            echo "Environment secret GITHUB_TOKEN is not set. Make sure you add a correct Github API token before running this pipeline."
+            exit 1
+          fi
+
+      - name: Check if backup environment exists
+        id: check-env
+        run: |
+          ENV_NAME="backup"
+          RESPONSE=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                            "https://api.github.com/repos/${{ github.repository }}/environments/$ENV_NAME")
+          if echo "$RESPONSE" | grep -q '"name": "'$ENV_NAME'"'; then
+            echo "Environment $ENV_NAME exists."
+            echo "::set-output name=exists::true"
+          else
+            echo "Environment $ENV_NAME does not exist."
+            echo "::set-output name=exists::false"
+          fi
+
+  fetch-production-credentials:
+    name: Fetch backup encryption key for production environment to be used in periodic restore
+    needs: check-production-environment
+    runs-on: ubuntu-22.04
+    environment: production
+    if: needs.check-production-environment.outputs.production_env_exists == 'true' && github.event.inputs.environment == 'staging'
+    outputs:
+      production-decryption-key: ${{ steps.fetch-credentials.outputs.production-decryption-key }}
+    steps:
+      - name: Fetch decryption key
+        id: fetch-credentials
+        run: |
+          BACKUP_ENCRYPTION_PASSPHRASE="${{ secrets.BACKUP_ENCRYPTION_PASSPHRASE }}"
+          echo -n "$BACKUP_ENCRYPTION_PASSPHRASE" | openssl enc -aes-256-cbc -pbkdf2 -salt -k "${{ secrets.GH_ENCRYPTION_PASSWORD }}" -out encrypted_key.bin
+          ENCODED_ENCRYPTED_SECRET=$(base64 < encrypted_key.bin)
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "production-decryption-key<<$EOF" >> $GITHUB_OUTPUT
+          echo "$ENCODED_ENCRYPTED_SECRET" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
```

Comment:
@n1koo Think we need an abstraction around passing secrets from one job to other. This would be so that we can read a secret from one environment (e.g. backend environment, SSH_KEY variable)

This could be one option
https://github.com/cloudposse/github-action-secret-outputs